cmd/podman/oci-trace-hook/trace.go

```diff
@@ -0,0 +1,238 @@
+package main
+
+import (
+	"bufio"
+	"bytes"
+	"encoding/binary"
+	"encoding/json"
+	"flag"
+	"io/ioutil"
+	"log"
+	"os"
+	"os/signal"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/docker/docker/api/types"
+	bpf "github.com/iovisor/gobpf/bcc"
+	spec "github.com/opencontainers/runtime-spec/specs-go"
+	seccomp "github.com/seccomp/libseccomp-golang"
+)
+
+type event struct {
+	Pid uint32
+	ID  uint32
+	// Inum    uint
+	Command [16]byte
+}
+
+type calls map[string]int
+
+const source string = `
+#include <linux/bpf.h>
+#include <linux/nsproxy.h>
+#include <linux/pid_namespace.h>
+#include <linux/ns_common.h>
+#include <linux/sched.h>
+#include <linux/tracepoint.h>
+
+BPF_HASH(parent_namespace, u64, unsigned int);
```

Comment:
Where do these come from? I don't see them implemented here.

Reply:
The `BPF_HASH()` comes from bcc https://github.com/iovisor/bcc/blob/master/docs/reference_guide.md#2-bpf_hash

Reply:
@weirdwiz, we need to document and comment the code a bit more. I prefer to have more verbose comments to make sure that readers will understand the intention and such details more quickly.

Reply:
yes, you're right I should add more comments in the code, fixing it

Reply:
Still valid.